src/GenerateUnionRecord/UnionRecordGenerator.cs

```diff
@@ -59,20 +66,18 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
     private static void Execute(
         Compilation compilation,
-        ImmutableArray<RecordDeclarationSyntax> recordDeclarations,
+        ImmutableArray<GenerationTarget> targets,
         SourceProductionContext context
     )
     {
-        if (recordDeclarations.IsDefaultOrEmpty)
+        if (targets.IsDefaultOrEmpty)
         {
             return;
         }
 
-        var distinctRecords = recordDeclarations.Distinct();
```

Comment:
Shouldn't be necessary. I think this was originally a defense against two unions with the same name but separate namespaces, but I don't think that should be an issue anymore.

However, we probably should add a test to ensure that isn't an issue to be sure. Feel free to add that here or in a separate PR if I don't get to it.

Reply:
Did some investigation here, and we have a bug where we cannot have two records with the same name in separate namespaces because the source generated file is named after the record's name. I'll file a bug for it and no need to address that here.